### ISSUE TITLE
Question regarding IoUs of pretrained HRNet Proto

### ISSUE THREAD

**Issue by ksakmann** [tfzhou/ProtoSeg#10]
Hi, I downloaded the checkpoint, prepared the data and ran the evaluation script : 
```
bash scripts/cityscapes/hrnet/run_h_48_d_4_proto.sh val hrnet_proto_80k
```
I had to include a tiny fix `label_img_ = Image.fromarray(label_img_)`  instead of  `label_img_ = Image.fromarray(label_img_, 'P')` in `tester.py` because the labels were all black in the output directory.  If I then execute the above line,  I end up with an mIoU of *85.7*,  much better than the  *81.1* reported in your paper in Table 2 for HRNet. This is the output: 
```
classes          IoU      nIoU
--------------------------------
road          : 0.978194      nan
sidewalk      : 0.817676      nan
building      : 0.966470      nan
wall          : 0.586037      nan
fence         : 0.650901      nan
pole          : 0.858629      nan
traffic light : 0.865478      nan
traffic sign  : 0.904490      nan
vegetation    : 0.977709      nan
terrain       : 0.664823      nan
sky           : 0.973433      nan
person        : 0.950269    0.000000
rider         : 0.802941    0.000000
car           : 0.985926    0.000000
truck         : 0.818478    0.000000
bus           : 0.939901    0.000000
train         : 0.811736    0.000000
motorcycle    : 0.806717    0.000000
bicycle       : 0.919648    0.000000
--------------------------------
Score Average : 0.856814    0.000000
--------------------------------
```
I also used my own evaluation script on the generated labels that are in the `label` directory and I  get exactly the same results. Could you check?

**Comment by tfzhou** [comment]
@ksakmann I guess that there are some issues about your testing. I remember that some guys reported the problem in other projects but I cannot recall the details. I will check and come back to you later.

**Comment by ksakmann** [comment]
> @ksakmann I guess that there are some issues about your testing....

I thought so too at first, but when I take the predicted labels  saved in the output `label` dir and run a different IoU evaluation script on them, I  arrive at the exact same numbers as above, i.e. an mIoU of 85.6% 

If you want to reproduce it: 
* in `tester.py` remove the palletized output Flag `'P'` in `fromarray(label_img_, 'P')`: 
``` 
208                    label_img_ = Image.fromarray(label_img_)
```
* set `CSUPPORT=False` in `cityscapes_evaluator.py` to use plain numpy.

**Comment by ksakmann** [comment]
Sorry for the confusion, my bad. I blindly copy pasted the instructions for data preparation from openseg.pytorch which have the flag `coarse` set to `True`. So the numbers I got were for the coarse labels.
